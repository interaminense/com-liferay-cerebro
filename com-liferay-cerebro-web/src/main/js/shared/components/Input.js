import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

import Button from './Button';

const SIZES = ['lg', 'sm'];

class InputAddon extends Component {
	render() {
		return <div class="input-group-addon">{this.props.children}</div>;
	}
}

class InputButton extends Component {
	render() {
		return (
			<span class="input-group-btn">
				<Button {...this.otherProps()}>{this.props.children}</Button>
			</span>
		);
	}
}

class InputInset extends Component {
	render() {
		return (
			<span class="input-group-inset-item">{this.props.children}</span>
		);
	}
}

class InputGroupInput extends Component {
	render() {
		return <div class="input-group-input">{this.props.children}</div>;
	}
}

class InputGroup extends Component {
	render() {
		const {children, inset, size} = this.props;

		const classes = getCN('input-group', {
			'input-group-inset': inset,
			[`input-group-${size}`]: size
		});

		return (
			<div class={classes} {...this.otherProps()}>
				{children}
			</div>
		);
	}
}

InputGroup.PROPS = {
	inset: Config.bool().value(false),
	size: Config.oneOf(SIZES)
};

class Input extends Component {
	/**
	 * Public method used for focusing the input element.
	 */
	focus() {
		this.element.focus();
	}

	/**
	 * Public method used for retrieving value of input.
	 */
	getValue() {
		return this.element.value;
	}

	render() {
		const {size} = this.props;

		const classes = getCN('input-root', 'form-control', {
			[`form-control-${size}`]: size
		});

		return <input type="text" {...this.otherProps()} class={classes} />;
	}
}

Input.PROPS = {
	size: Config.oneOf(SIZES)
};

Input.Addon = InputAddon;
Input.Group = InputGroup;
Input.GroupInput = InputGroupInput;
Input.Button = InputButton;
Input.Inset = InputInset;

Input.SIZES = SIZES;

export default Input;