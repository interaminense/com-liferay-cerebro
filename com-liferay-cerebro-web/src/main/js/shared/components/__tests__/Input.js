import Input from '../Input';

describe('Input', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render input', () => {
		component = new Input();

		expect(component).toMatchSnapshot();
	});

	it('should render input with a different size', () => {
		component = new Input({size: 'lg'});

		expect(component).toMatchSnapshot();
	});

	it('should render input group', () => {
		component = new Input.Group();

		expect(component).toMatchSnapshot();
	});

	it('should render input group with different size', () => {
		component = new Input.Group({size: 'lg'});

		expect(component).toMatchSnapshot();
	});

	it('should render input group with inset', () => {
		component = new Input.Group({inset: true});

		expect(component).toMatchSnapshot();
	});

	it('should render input group input', () => {
		component = new Input.GroupInput();

		expect(component).toMatchSnapshot();
	});

	it('should render input inset', () => {
		component = new Input.Inset();

		expect(component).toMatchSnapshot();
	});

	it('should render input button', () => {
		component = new Input.Button();

		expect(component).toMatchSnapshot();
	});

	it('should render input addon', () => {
		component = new Input.Addon();

		expect(component).toMatchSnapshot();
	});
});