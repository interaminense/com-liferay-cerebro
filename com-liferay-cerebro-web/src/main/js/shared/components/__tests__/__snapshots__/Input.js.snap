// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render input 1`] = `<input class="input-root form-control" type="text">`;

exports[`Input should render input addon 1`] = `<div class="input-group-addon"></div>`;

exports[`Input should render input button 1`] = `
<span class="input-group-btn">
  <button class="button-root btn btn-secondary" type="button"></button>
</span>
`;

exports[`Input should render input group 1`] = `<div class="input-group"></div>`;

exports[`Input should render input group input 1`] = `<div class="input-group-input"></div>`;

exports[`Input should render input group with different size 1`] = `<div class="input-group input-group-lg"></div>`;

exports[`Input should render input group with inset 1`] = `<div class="input-group input-group-inset"></div>`;

exports[`Input should render input inset 1`] = `<span class="input-group-inset-item"></span>`;

exports[`Input should render input with a different size 1`] = `<input class="input-root form-control form-control-lg" type="text">`;
